tests/e2e/etcdmigration/etcdmigration_test.go

```diff
@@ -0,0 +1,214 @@
+		It("Starts K3s with SQLite datastore", func() {
+			err := StartK3sWithSQLite(tc.AllNodes())
+			Expect(err).NotTo(HaveOccurred(), e2e.GetVagrantLog(err))
+
+			By("CLUSTER CONFIG")
+			By("OS:" + *nodeOS)
+			By(tc.Status())
+			tc.KubeconfigFile, err = e2e.GenKubeconfigFile(tc.Servers[0].String())
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		It("Checks node and pod status with SQLite", func() {
+			By("Fetching node status")
+			Eventually(func() error {
+				return tests.NodesReady(tc.KubeconfigFile, e2e.VagrantSlice(tc.AllNodes()))
+			}, "600s", "5s").Should(Succeed())
+			Eventually(func() error {
+				return tests.AllPodsUp(tc.KubeconfigFile)
+			}, "600s", "5s").Should(Succeed())
+			Eventually(func() error {
+				return tests.CheckDefaultDeployments(tc.KubeconfigFile)
+			}, "480s", "10s").Should(Succeed())
+			e2e.DumpPods(tc.KubeconfigFile)
+		})
+
+		It("Creates test resources before migration", func() {
+			// Create a test ConfigMap to verify data persistence
+			createCmd := "kubectl create configmap migration-test --from-literal=test=before-migration"
+			_, err := tc.Servers[0].RunCmdOnNode(createCmd)
+			Expect(err).NotTo(HaveOccurred())
+
+			// Verify the ConfigMap exists
+			getCmd := "kubectl get configmap migration-test -o jsonpath='{.data.test}'"
+			result, err := tc.Servers[0].RunCmdOnNode(getCmd)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(result).To(ContainSubstring("before-migration"))
+		})
+
+		It("Migrates from SQLite to etcd", func() {
+			err := MigrateToEtcd(tc.Servers[0])
+			Expect(err).NotTo(HaveOccurred(), e2e.GetVagrantLog(err))
+
+			// Verify etcd is running
+			Eventually(func() (string, error) {
+				return tc.Servers[0].RunCmdOnNode("systemctl status k3s | grep 'Started k3s'")
+			}, "60s", "5s").Should(ContainSubstring("Started k3s"))
+
+			// Verify etcd is being used
+			Eventually(func() (string, error) {
+				return tc.Servers[0].RunCmdOnNode("journalctl -u k3s --no-pager | grep -i etcd")
+			}, "120s", "5s").Should(ContainSubstring("etcd"))
+		})
+
+		It("Checks node and pod status after migration", func() {
+			By("Fetching node status after migration")
+			Eventually(func() error {
+				return tests.NodesReady(tc.KubeconfigFile, e2e.VagrantSlice(tc.AllNodes()))
+			}, "600s", "5s").Should(Succeed())
+			Eventually(func() error {
+				return tests.AllPodsUp(tc.KubeconfigFile)
+			}, "600s", "5s").Should(Succeed())
+			Eventually(func() error {
+				return tests.CheckDefaultDeployments(tc.KubeconfigFile)
+			}, "480s", "10s").Should(Succeed())
+			e2e.DumpPods(tc.KubeconfigFile)
+		})
+
+		It("Verifies data persistence after migration", func() {
+			// Verify the ConfigMap still exists with the same data
+			getCmd := "kubectl get configmap migration-test -o jsonpath='{.data.test}'"
+			result, err := tc.Servers[0].RunCmdOnNode(getCmd)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(result).To(ContainSubstring("before-migration"))
+		})
+
+		It("Kills the cluster", func() {
```

Comment:
Nit: no need to kill the cluster, the next step is to cleanup and destroy everything in AfterSuite.